controllers/component_build_controller.go

```diff
@@ -126,34 +145,176 @@ func (r *ComponentBuildReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		return ctrl.Result{}, nil
 	}
 
-	// Set initial build annotation to prevent next builds
-	component.Annotations[InitialBuildAnnotationName] = "true"
-	if err := r.Client.Update(ctx, &component); err != nil {
-		return ctrl.Result{}, err
-	}
+	if val, ok := component.Annotations[gitops.PaCAnnotation]; ok && val == "1" {
+		r.Log.Info("PaC enabled")
+		if err := r.EnsurePACRepository(ctx, component); err != nil {
+			return ctrl.Result{}, err
+		}
+		if err := r.GeneratePullRequest(ctx, component); err != nil {
+			return ctrl.Result{}, err
+		}
 
-	if err := r.SubmitNewBuild(ctx, component); err != nil {
-		// Try to revert the annotation
-		if err := r.Client.Get(ctx, req.NamespacedName, &component); err == nil {
-			component.Annotations[InitialBuildAnnotationName] = "false"
-			if err := r.Client.Update(ctx, &component); err != nil {
+		// Set initial build annotation to prevent next builds
+		component.Annotations[InitialBuildAnnotationName] = "true"
+		if err := r.Client.Update(ctx, &component); err != nil {
+			return ctrl.Result{}, err
+		}
+	} else {
+		// Set initial build annotation to prevent next builds
+		component.Annotations[InitialBuildAnnotationName] = "true"
+		if err := r.Client.Update(ctx, &component); err != nil {
+			return ctrl.Result{}, err
+		}
+
+		if err := r.SubmitNewBuild(ctx, component); err != nil {
+			// Try to revert the annotation
+			if err := r.Client.Get(ctx, req.NamespacedName, &component); err == nil {
+				component.Annotations[InitialBuildAnnotationName] = "false"
```

Comment:
also, it seems like GeneratePipelineRun creates several annotations.  Should we not clean up all of them if this is in fact a"retry as normal / not as pipeline as code" path.